lib/main.dart

```diff
@@ -158,6 +149,47 @@ class _WithSelectedNetworkState extends State<_WithSelectedNetwork> {
   }
 }
 
+class _WithTermsAndConditionAcceptance extends StatefulWidget {
+  final Widget child;
+
+  const _WithTermsAndConditionAcceptance({required this.child});
+
+  @override
+  State<_WithTermsAndConditionAcceptance> createState() => _WithTermsAndConditionAcceptanceState();
+}
+
+class _WithTermsAndConditionAcceptanceState extends State<_WithTermsAndConditionAcceptance> {
+  late final Future<FutureValue<AcceptedTermsAndConditionsState?>> _lastAccepted;
+  late final TermsAndConditionsRepository _repository;
+
+  @override
+  void initState() {
+    super.initState();
+    final storage = context.read<ServiceRepository>().storage;
+    setState(() {
+      _repository = TermsAndConditionsRepository(storageProvider: storage);
+      _lastAccepted = _repository.getAcceptedTermsAndConditions().then((value) => FutureValue(value));
+    });
+  }
+
+  @override
+  Widget build(BuildContext context) {
+    return FutureBuilder(
+        future: _lastAccepted,
+        builder: (_, snapshot) {
+          // Check on the connection state since last accepted possible null
```

Comment:
This comment is no longer relevant. I also believe we should use the helper functions provided by the library and do it as:
```
if (snapshot.hasData) {
   return BlockProvider(... return TermsAndConditionAcceptance(_repository, snapshot.requireData.value));
} else if (snapshot.hasError) {
  // TODO Handle the error.
}
return const _Initializing();
```
As it is now, an error thrown by `_lastAccepted` will result in being stuck in `_Initializing`. So I only think it is safe to not make the `hasError` check if the function called cannot throw. And even then if it is changed in the future into something that throws, then you will have the same issue.